kubernetes/deployment-production.tmpl

```diff
@@ -1,96 +1,72 @@
 apiVersion: v1
 data:
   nginx.conf: |+
-    upstream docker-panoptes {
-      server localhost:81;
-    }
+    gzip_types *;
 
-    server {
-      server_name panoptes.zooniverse.org;
-      include /etc/nginx/ssl.default.conf;
-      gzip_types *;
+    proxy_buffer_size 8k;
+    proxy_set_header Host $http_host;
+    proxy_set_header X-Real-IP $remote_addr;
 
-      proxy_buffer_size 8k;
-      proxy_set_header Host $http_host;
-      proxy_set_header X-Real-IP $remote_addr;
+    location = /commit_id.txt {
+      root /static-assets/;
+      expires off;
+      add_header 'Access-Control-Allow-Origin' '*';
+      add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS, HEAD';
+      add_header 'Access-Control-Allow-Credentials' 'true';
+    }
 
-      location = /commit_id.txt {
-        root /static-assets/;
-        expires off;
-        add_header 'Access-Control-Allow-Origin' '*';
-        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS, HEAD';
-        add_header 'Access-Control-Allow-Credentials' 'true';
-      }
+    location ~ ^/api-assets/ {
+      root /static-assets/;
+      gzip_static on; # to serve pre-gzipped version
+      expires max;
+      add_header Cache-Control public;
+    }
 
-      location ~ ^/api-assets/ {
-        root /static-assets/;
-        gzip_static on; # to serve pre-gzipped version
-        expires max;
-        add_header Cache-Control public;
-      }
+    location = /favicon.ico {
+      root /static-assets/;
+      expires max;
+      add_header Cache-Control public;
+    }
 
-      location = /favicon.ico {
-        root /static-assets/;
-        expires max;
-        add_header Cache-Control public;
-      }
+    location = /robots.txt {
+      root /static-assets/;
+      expires max;
+      add_header Cache-Control public;
+    }
 
-      location = /robots.txt {
-        root /static-assets/;
-        expires max;
-        add_header Cache-Control public;
-      }
+    location / {
+      # avoid attacker setting the host value for links in rails_app
+      # https://github.com/rails/rails/issues/29893
+      proxy_set_header X-Forwarded-Host $server_name;
+      proxy_pass http://docker-panoptes;
+    }
+kind: ConfigMap
+metadata:
+  name: panoptes-nginx-common-conf-production
+---
+apiVersion: v1
+data:
+  nginx.conf: |+
+    upstream docker-panoptes {
+      server localhost:81;
+    }
+    
+    server {
+      server_name www.zooniverse.org;
```

Comment:
If these server blocks are all the same you don't really need more than one. You can group multiple names together:

```
      server_name www.zooniverse.org panoptes.zooniverse.org signin.zooniverse.org;
```

Reply:
my reading suggests that this will use the `www.zooniverser.org` as the $server_name variable which will then be passed to the upstream via host header, https://github.com/zooniverse/panoptes/blob/325d71760298c701efa5db1202e8b5b4232112b1/kubernetes/deployment-production.tmpl#L40

this will break redirects as per the issue outlined in https://github.com/zooniverse/operations/issues/459

If we can verify how the $server_name variable is filled when using one server block (wildcard subdomain / regex are options here) then i'm happy to collapse them - i did look into collapsing them but couldn't ensure the $server_name var will be correctly set.

Reply:
In my testing it seems to set `$server_name` to the first hostname value:

```nginx
server {
	server_name test testing;
	rewrite ^.*$ http://$server_name redirect;
}
```

```
$ curl -v -H "Host: test" localhost:8080/ | grep Location
< Location: http://test
$ curl -v -H "Host: testing" localhost:8080/ | grep Location
< Location: http://test
```

So I guess that's not what we want.

We'll need to keep them separated as you have done then in that case. (Another option might be to use `$http_host` but that could be dangerous if we accidentally deploy with no default server block, as that would open it to every host name.)

Reply:
thanks for doing the actual testing Adam - appreciated. FWIW the security issue is fixed in rails 6+ so we can remove this nginx header config if / when we get there.